default.nix

```diff
@@ -1,17 +1,64 @@
-with import (builtins.fetchGit {
-  url = https://github.com/NixOS/nixpkgs-channels;
-  ref = "nixos-18.09-small";
-  rev = "95fed28ac372c61eb83c87ad97c24b0f957827bf";
-}) {};
+let
+  # Commit of the Nixpkgs repository that we want to use.
+  nixpkgsVersion = {
+      date = "2020-10-04";
+      rev = "d191ee22f0b0c91070f9f4c79210d4e398df0b93";
+      tarballHash = "13fb4cgxxc47shmzbm0rawfxjnilf7r4zq1vwc0z6s3lxijl9wl9";
+    };
+
+  # Nix files that describe the Nixpkgs repository. We evaluate the expression
+  # using `import` below.
+  pkgs = import (fetchTarball {
+      url = "https://github.com/nixos/nixpkgs/archive/${nixpkgsVersion.rev}.tar.gz";
+      sha256 = nixpkgsVersion.tarballHash;
+    }) {};
+
+  python = pkgs.python36.withPackages (ps: [ps.sphinx ps.sphinx_rtd_theme ps.livereload]);
```

Comment:
Good job, not too obvious and easy to figure that out :-)

Reply:
I wonder whether that creates the caching situation? Does that not effectively create a python derivation with exactly those packages? So to have caching for that, somebody else would have to do the exact same thing...?

I will try your suggestions from above of course.

Reply:
my bet is on the 3.6 Python version

Reply:
You seem to know your craft... newer python worked :)